modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,148 +558,187 @@ sub print_diagram {
 sub apply_tweaks {
     my $self    = shift @_;
     my $tweaks  = shift @_;
-
+    my @response;
+    my $responseStructure;
     my $need_write = 0;
-
+    $responseStructure->{Tweaks} = [];
     foreach my $tweak (@$tweaks) {
-        print "\nTweak.Request\t$tweak\n";
+        push @response, "\nTweak.Request\t$tweak\n";
 
         if($tweak=~/^pipeline\.param\[(\w+)\](\?|#|=(.+))$/) {
             my ($param_name, $operator, $new_value_str) = ($1, $2, $3);
-
+            my $tweakStructure;
+            $tweakStructure->{Action} = $self->{ACTION}->{substr($operator, 0, 1)};
+            $tweakStructure->{Object}->{Type} = 'Pipeline';
```

Comment:
What about defining all the object types in a separate hash like the error messages and the actions ? This would guarantee consistency across the various regexp each type accepts